ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -516,10 +515,10 @@ loadSessionWithOptions recorder SessionLoadingOptions{..} dir = do
                             HM.unionWith (<>) k $ HM.fromList knownTargets
                 hasUpdate = if known /= known' then Just (unhashed known') else Nothing
             writeTVar knownTargetsVar known'
-            logDirtyKeys <- recordDirtyKeys extras GetKnownTargets [emptyFilePath]
-            return (logDirtyKeys >> pure hasUpdate)
+            return (pure hasUpdate)
```

Comment:
I think you can remove the `join` wrapping this block if you remove this extraneous return or pure

```suggestion
            pure hasUpdate
```

Reply:
done